src/artifactorypurge/artifactorypurge.py

```diff
@@ -1,64 +1,78 @@
-    artifactory = Artifactory(credentials['artifactory_url'], 
-                              credentials['artifactory_username'],
+    artifactory = Artifactory(credentials['artifactory_url'], credentials['artifactory_username'],
                               credentials['artifactory_password'])
 
-    before = artifactory.list(reponame)
+    plugin_source = setup_pluginbase(plugin_path)
+    before = artifactory.list(None)
     for repo, info in before.items():
-        fp = 0
+        plugin_name = repo.replace("-", "_")
         try:
-            modulename = repo.replace("-", "_")
-            fp, pathname, description = imp.find_module(modulename, ["repositories",])
-            module =  imp.load_module(modulename, fp, pathname, description)
-            artifacts = module.purgelist(artifactory, repo, project)
-            count = artifactory.purge(repo, dryrun, artifacts)
-            LOG.info("processed {}, purged {}".format(repo, count))
-        except IndexError as e:  # FIXME: return to generic catch
-            exceptions[repo] = str(e)
-        except ImportError:
-            continue
-        finally:
-            if fp:
-                fp.close()
+            artifactory_plugin = plugin_source.load_plugin(plugin_name)
+        except ModuleNotFoundError:
+            if default:
+                LOG.info("No plugin found for %s. Applying Default", repo)
+                artifactory_plugin = plugin_source.load_plugin('default')
+            else:
+                LOG.info("No plugin found for %s. Skipping Default", repo)
+                continue
+        artifacts = artifactory_plugin.purgelist(
+            artifactory,
+            repo,
+            None,
+        )
+        count = artifactory.purge(repo, dryrun, artifacts)
+        LOG.info("processed {}, purged {}".format(repo, count))
 
     LOG.info("")
     LOG.info("Purging Performance:")
-    after = artifactory.list(reponame)
+    after = artifactory.list(None)
     for repo, info in after.items():
         try:
             get_performance_report(repo, before[repo], info)
         except IndexError:
             pass
 
-    if len(exceptions):
-        LOG.error("There were errors:")
-        for repo, exception in exceptions.items():
-            LOG.error("  {}: {}".format(repo, exception))
-
-        exit(1)
-
     LOG.info("Done.")
 
     exit(0)
 
 
+def setup_pluginbase(plugin_path):
+    """Sets up plugin base with default path and provided path
+    
+    Args:
+        plugin_path (str): Extra path to find plugins in
+
+    Returns:
+        PluginSource: PluginBase PluginSource for finding plugins
+    """
+    here = os.path.dirname(os.path.realpath(__file__))
+    default_path = "{}/plugins".format(here)
+    LOG.info("Searching for plugins in %s and %s", plugin_path, default_path)
+    plugin_base = PluginBase(package='artifactorypurge.plugins')
+    plugin_source = plugin_base.make_plugin_source(searchpath=[plugin_path, default_path])
```

Comment:
Once `extra_plugin_path` is optional, you'll have to adjust this.
Maybe soemthing like this:
```
+    all_paths = [default_path]
+    if plugin_path:
+        all_paths.append(plugin_path)
```